app/src/main/java/org/mozilla/focus/telemetry/SentryIntegration.kt

```diff
@@ -50,6 +50,39 @@ object SentryWrapper {
         // they're personally identifiable information.
         Sentry.getContext().user = SentryUnknownUser()
     }
+
+    /**
+     * One way to test calls to static methods is to pass instances of their objects into the
+     * class being tested. This allows tests to pass spies or mocks instead.
+     *
+     * Example:
+     *
+     *  class SimpleClass(private val singleton: Singleton = Singleton) {
+     *      fun testThis() {
+     *          singleton.callMethod()
+     *      }
+     *  }
+     *
+     *  class SimpleClassTest {
+     *
+     *      @Test
+     *      fun testSimpleClass() {
+     *          val spy = spy(Singleton)
+     *          val simpleClass = SimpleClass(spy)
+     *          simpleClass.testThis()
+     *          verify(spy).callMethod()
+     *      }
+     *  }
+     *
+     * [Sentry] has a private constructor, disallowing this pattern, so instead we delegate
```

Comment:
Another reason we force everyone to go through `SentryIntegration` (same for Telemetry) is modularity: if we need to swap out the implementation of the Integration class, consumers can make the same calls. No one in the app code should know about `Sentry`, just `SentryIntegration`.

Since this is the primary reason for having SentryIntegration, I don't think this comment is necessary.

I don't know where to comment about why we do these `Integration` classes like that – maybe a README in an `integrations` package?

Reply:
Hm.  I'm not sure if we need to include this in the project at all, now that you mention it.  What do you think about adding a page in our shared documentation repo?  I could take the time to write one.

Reply:
Good idea. Maybe we should link to it from each of the `Integration` pages, just so it's not overlooked? Like:
```
/* License
 */

// See <url> for dependency integration best practices.

class TelemetryIntegration {

Reply:
I've made a note to myself to write it, and I'll open a new PR adding the link once it's done.